tidy3d/components/structure.py

```diff
@@ -93,8 +94,25 @@ def eps_diagonal(self, frequency: float, coords: Coords) -> Tuple[complex, compl
         """
         if isinstance(self.medium, CustomMedium):
             return self.medium.eps_diagonal_on_grid(frequency=frequency, coords=coords)
+        if isinstance(self.medium, Medium2D):
+            raise ValidationError(
+                "Call volumetric_equivalent before evaluating permittivity of a Medium2D."
```

Comment:
Regular single quotes (apostrophes) `'` around variable names in error messages / warnings.
```
Call 'Simulation.volumetric_equivalent' before evaluating permittivity of a 'Medium2D'.
```